src/tree/mod.rs

```diff
@@ -737,122 +749,69 @@ impl RatchetTree {
             self.nodes.truncate(new_tree_size);
         }
     }
-    /// Computes the tree hash
-    pub fn compute_tree_hash(&self) -> Vec<u8> {
-        fn node_hash(ciphersuite: &Ciphersuite, tree: &RatchetTree, index: NodeIndex) -> Vec<u8> {
-            let node = &tree.nodes[index.as_usize()];
-            match node.node_type {
-                NodeType::Leaf => {
-                    let leaf_node_hash = LeafNodeHashInput::new(&index, &node.key_package);
-                    leaf_node_hash.hash(ciphersuite)
-                }
-                NodeType::Parent => {
-                    let left = treemath::left(index)
-                        .expect("node_hash: Error when computing left child of node.");
-                    let left_hash = node_hash(ciphersuite, tree, left);
-                    let right = treemath::right(index, tree.leaf_count())
-                        .expect("node_hash: Error when computing left child of node.");
-                    let right_hash = node_hash(ciphersuite, tree, right);
-                    let parent_node_hash = ParentNodeHashInput::new(
-                        index.as_u32(),
-                        &node.node,
-                        &left_hash,
-                        &right_hash,
-                    );
-                    parent_node_hash.hash(ciphersuite)
-                }
-                NodeType::Default => panic!("Default node type not supported in tree hash."),
+
+    /// Verify the parent hash of a tree node. Returns `true` if the parent
+    /// hash has successfully been verified and `false` otherwise.
+    pub fn verify_parent_hash(&self, index: NodeIndex, node: &Node) -> bool {
+        // "Let L and R be the left and right children of P, respectively"
+        let left = treemath::left(index).unwrap();
+        let right = treemath::right(index, self.leaf_count()).unwrap();
+        // Unwrapping here is safe, because we know it is a full parent node
+        let parent_hash_field = &node.parent_hash().unwrap();
```

Comment:
I'm not sure we can say that we know that. The function is public and just takes a `&Node` and an `NodeIndex`. We don't know if it's actually a parent node (or if the unwrapping of the left and right child nodes above will panic for that matter).